atom/browser/atom_browser_client.cc

```diff
@@ -140,8 +185,7 @@ void AtomBrowserClient::OverrideSiteInstanceForNavigation(
     return;
   }
 
-  // Restart renderer process for all navigations except "javacript:" scheme.
-  if (url.SchemeIs(url::kJavaScriptScheme))
+  if (!ShouldCreateNewSiteInstance(browser_context, current_instance, url))
```

Comment:
What do you think about patching libchromiumcontent to notify `AtomBrowserClient` whenever a new `RenderProcessHost` is created?

I suggest we add a new method(eg: `ContentBrowserClient::OnProcessCreatedForNavigation`) that is called called after the `RenderProcessHost` is created but before the process is launched(and passed arguments similar to `OverrideSiteInstanceForNavigation`), so we have the process ID and can populate `pending_process_` before `AppendExtraCommandLineSwitches` is called.

If you agree I can send a PR to https://github.com/electron/libchromiumcontent